lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2642,26 +2639,13 @@ void TypeChecker::checkDeclAttributes(Decl *D) {
   }
 }
 
-void TypeChecker::checkTypeModifyingDeclAttributes(VarDecl *var) {
-  if (!var->hasType())
-    return;
-
-  if (auto *attr = var->getAttrs().getAttribute<ReferenceOwnershipAttr>())
-    checkReferenceOwnershipAttr(var, attr);
-}
-
-void TypeChecker::checkReferenceOwnershipAttr(VarDecl *var,
+Type TypeChecker::checkReferenceOwnershipAttr(VarDecl *var, Type type,
```

Comment:
Can you rename this too, to indicate that it has a return type?

Reply:
I have another change coming up that moves this to TypeCheckStorage along with the code for setting the interface type of a VarDecl, instead of setting it as a side effect of coercePatternToType(). I haven't renamed it yet though. Do you mind if I come up with the better name in the follow-up PR?

Reply:
Sure, that works.